### ISSUE TITLE
Iced applications fail to compile under wasm target on master

### ISSUE THREAD

**Issue by Gohla** [hecrj/iced#129]
On current master, Iced applications fail to compile under the wasm target, because `Subscription` is not implemented in `iced_web`:

```
error[E0432]: unresolved import `crate::Subscription`
 --> C:\Dev\iced\src\application.rs:1:41
  |
1 | use crate::{Command, Element, Settings, Subscription};
  |                                         ^^^^^^^^^^^^ no `Subscription` in the root

error[E0432]: unresolved import `crate::Subscription`
 --> C:\Dev\iced\src\sandbox.rs:1:54
  |
1 | use crate::{Application, Command, Element, Settings, Subscription};
  |                                                      ^^^^^^^^^^^^ no `Subscription` in the root
```

Maybe there could be a wasm build in the GitHub actions to highlight these issues?

**Comment by hecrj** [comment]
Given we are still exploring, I don't think we should currently focus on offering a fully compatible cross-platform API when it comes to widgets in `master`. We should be able to batch work on each side and avoid jumping back and forth too much.

However, I agree `iced` itself should always compile when targetting Wasm. I fixed this in #131 and added a check in the CI pipeline.

**Comment by Gohla** [comment]
Right, it's actually `iced` that failed to compile on wasm, not necessarily an iced application. I agree about not having compatibility on master, that would be too limiting. Thanks for fixing and adding the check!
